RUFAS/biophysical/animal/pen.py

```
@@ -596,6 +604,92 @@ def get_manure_data(self) -> PenManureData:
+        """
+        animal_manure_streams: list[dict[str, ManureStream]] = []
+
+        total_pen_manure_data = NewPenManureData(
+            num_animals=len(self.animals_in_pen),
+            manure_deposition_surface_area=1.0,
+            animal_combination=self.animal_combination,
+            pen_type=self.pen_type,
+            manure_urine_mass=self.total_manure_excretion.urine,
+            manure_urine_nitrogen=self.total_manure_excretion.urine_nitrogen,
+            stream_type=StreamType.GENERAL,
+        )
+
+        total_stream = ManureStream(
+            water=self.total_manure_excretion.manure_mass - self.total_manure_excretion.total_solids,
+            ammoniacal_nitrogen=self.total_manure_excretion.manure_total_ammoniacal_nitrogen,
+            nitrogen=self.total_manure_excretion.manure_nitrogen,
+            phosphorus=self.total_manure_excretion.phosphorus,
+            potassium=self.total_manure_excretion.potassium,
+            ash=0,
+            non_degradable_volatile_solids=self.total_manure_excretion.non_degradable_volatile_solids,
+            degradable_volatile_solids=self.total_manure_excretion.degradable_volatile_solids,
+            total_solids=self.total_manure_excretion.total_solids,
+            volume=self.total_manure_excretion.manure_mass / ManureConstants.MANURE_DENSITY,
+            pen_manure_data=total_pen_manure_data,
+        )
+
+        if self.animal_combination == AnimalCombination.LAC_COW:
+            parlor_stream_proportion = self.minutes_away_for_milking / 1440
+            general_stream_proportion = 1 - parlor_stream_proportion
+            manure_stream = total_stream.split_stream(
+                split_ratio=parlor_stream_proportion,
+                stream_type=StreamType.PARLOR,
+            )
+            if manure_stream.pen_manure_data is not None:
+                manure_stream.pen_manure_data.set_first_processor(self.parlor_stream_assignment)
```

Is this check necessary?
In other words, is it possible for a parlor manure stream to not have a "pen_manure_data" attribute?
Or should we raise an error if the "pen_manure_data" is None?

---

It was required to address a mypy error because `pen_manure_data` for a stream could be None (as is defined in the class).

---

I did try rearranging the `split_stream()` method to expressly return a manure stream with a `pen_manure_data` attribute that was not None but it didn't make a difference. I think we'd have to create an entirely separate ManureStream that didn't have the optional None for its `pen_manure_data` and that seems way too complicated.